packages/appsync-modelgen-plugin/src/utils/process-connections-v2.ts

```diff
@@ -0,0 +1,103 @@
+import { CodeGenField, CodeGenFieldDirective, CodeGenModel, CodeGenModelMap } from '../visitors/appsync-visitor';
+import {
+  CodeGenFieldConnection, DEFAULT_HASH_KEY_FIELD, flattenFieldDirectives,
+  getDirective,
+  makeConnectionAttributeName,
+} from './process-connections';
+import { processHasOneConnection } from './process-has-one';
+import { processBelongsToConnection } from './process-belongs-to';
+import { processHasManyConnection } from './process-has-many';
+
+// TODO: This file holds several references to utility functions in the v1 process connections file, those functions need to go here before that file is removed
+
+export function getConnectedFieldV2(field: CodeGenField, model: CodeGenModel, connectedModel: CodeGenModel, directiveName: string): CodeGenField {
+  const connectionInfo = getDirective(field)(directiveName);
+  if (!connectionInfo) {
+    throw new Error(`The ${field.name} on model ${model.name} is not connected`);
+  }
+
+  const connectionName = connectionInfo.arguments.name;
+  const indexName = connectionInfo.arguments.indexName;
+  const connectionFields = connectionInfo.arguments.fields;
+  if (connectionFields) {
+    let keyDirective;
+    if (indexName) {
+      keyDirective = flattenFieldDirectives(connectedModel).find(dir => {
+        return dir.name === 'index' && dir.arguments.name === indexName;
+      });
+      if (!keyDirective) {
+        throw new Error(
+          `Error processing @${connectionInfo.name} directive on ${model.name}.${field.name}, @index directive with name ${indexName} was not found in connected model ${connectedModel.name}`,
+        );
+      }
+    } else {
+      keyDirective = flattenFieldDirectives(connectedModel).find(dir => {
+        return dir.name === 'primaryKey';
+      });
+    }
+
+    // when there is a fields argument in the connection
+    const connectedFieldName = keyDirective ? ((fieldDir: CodeGenFieldDirective) => { return fieldDir.fieldName ;})(keyDirective as CodeGenFieldDirective) : DEFAULT_HASH_KEY_FIELD;
+
+    // Find a field on the other side which connected by a @connection and has the same fields[0] as indexName field
+    const otherSideConnectedField = connectedModel.fields.find(f => {
+      return f.directives.find(d => {
+        return (d.name === 'belongsTo' || d.name === 'hasOne' || d.name === 'hasMany') && d.arguments.fields && d.arguments.fields[0] === connectedFieldName;
+      });
+    });
+    if (otherSideConnectedField) {
+      return otherSideConnectedField;
+    }
+    // If there are no field with @connection with indexName then try to find a field that has same name as connection name
+    const connectedField = connectedModel.fields.find(f => f.name === connectedFieldName);
+
+    if (!connectedField) {
+      throw new Error(`Can not find key field ${connectedFieldName} in ${connectedModel}`);
+    }
+    return connectedField;
+  } else if (connectionName) {
```

Comment:
This condition is invalid since no argument `name` defined in v2

Reply:
Are you sure? I think this one is still valid because it's talking about the name of a connected field, not the name on a connection directive. It's for the fields argument

Reply:
This variable is defined as the latter one. In line 19: 
```TypeScript
const connectionName = connectionInfo.arguments.name;
```